pennylane/queuing.py

```diff
@@ -246,7 +250,126 @@ def _get_info(self, obj):
+
+
+def apply(op, context=QueuingContext):
+    """Apply an instantiated operator or measurement to a queuing context.
+
+    Args:
+        op (.Operator or .MeasurementProcess): the operator or measurement to apply/queue
+        context (.QueuingContext): The queuing context to queue the operator to.
+            Note that if no context is specified, the operator is
+            applied to currently active queuing context.
+
+    Returns:
+        .Operator or .MeasurementProcess: the input operator is returned for convenience
+
+    **Example**
+
+    In PennyLane, operations and measurements are 'queued' or applied to a QNode
+    when they are instantiated.
+
+    The ``apply`` function can be used to add operations that might have
+    already been instantiated elsewhere to the QNode:
+
+    .. code-block:: python
+
+        op = qml.RX(0.4, wires=0)
+        dev = qml.device("default.qubit", wires=2)
+
+        @qml.qnode(dev)
+        def circuit(x):
+            qml.RY(x, wires=0)
+            qml.apply(op)
+            return qml.expval(qml.PauliZ(0))
+
+    >>> print(qml.draw(circuit)(0.6))
+    0: ──RY(0.6)──RX(0.4)──┤ ⟨Z⟩
+
+    It can also be used to apply functions repeatedly:
+
+    .. code-block:: python
+
+        @qml.qnode(dev)
+        def circuit(x):
+            qml.apply(op)
+            qml.RY(x, wires=0)
+            qml.apply(op)
+            return qml.expval(qml.PauliZ(0))
+
+    >>> print(qml.draw(circuit)(0.6))
+    0: ──RX(0.4)──RY(0.6)──RX(0.4)──┤ ⟨Z⟩
+
+    .. UsageDetails::
+
+        Instantiated measurements can also be applied to queuing contexts
+        using ``apply``:
+
+        .. code-block:: python
+
+            meas = qml.expval(qml.PauliZ(0) @ qml.PauliY(1))
+            dev = qml.device("default.qubit", wires=2)
+
+            @qml.qnode(dev)
+            def circuit(x):
+                qml.RY(x, wires=0)
+                qml.CNOT(wires=[0, 1])
+                return qml.apply(meas)
+
+        >>> print(qml.draw(circuit)(0.6))
+         0: ──RY(0.6)──╭C──╭┤ ⟨Z ⊗ Y⟩
+         1: ───────────╰X──╰┤ ⟨Z ⊗ Y⟩
+
+        By default, ``apply`` will queue operators to the currently
+        active queuing context.
+
+        When working with low-level queuing contexts such as quantum tapes,
+        the desired context to queue the operation to can be explicitly
+        passed:
+
+        .. code-block:: python
+
+            with qml.tape.QuantumTape() as tape1:
+                qml.Hadamard(wires=1)
+
+                with qml.tape.QuantumTape() as tape2:
+                    op1 = qml.PauliX(wires=0)
+
+                    # apply the same operation to tape1
+                    # without leaving the tape2 context
+                    qml.apply(op1, context=tape1)
+
+                    qml.RZ(0.2, wires=0)
+
+                qml.CNOT(wires=[0, 1])
+
+        >>> tape1.operations
+        [Hadamard(wires=[1]), <QuantumTape: wires=[0], params=1>, PauliX(wires=[0]), CNOT(wires=[0, 1])]
```

Comment:
Why is the `<QuantumTape: wires=[0], params=1>` part there? It looks like only a `Hadamard`, `PauliX`, and `CNOT` should be applied to `tape1`?

Reply:
Oh wait I think I just understood why from the tests later on; the contents of `tape2` are also getting added to `tape1` because it's within the same queuing context, but using `qml.apply` allows us to apply something to `tape1` without also applying it to `tape2`?

Reply:
Yep exactly!